Firestore/Source/Remote/FSTRemoteEvent.mm

```diff
@@ -160,12 +155,8 @@ - (NSUInteger)hash {
 }
 
 - (void)applyTo:(DocumentKeySet *)keys {
-  for (const auto &key : _addedDocuments) {
-    keys->insert(key);
-  };
-  for (const auto &key : _removedDocuments) {
-    keys->erase(key);
-  };
+  keys->insert(_addedDocuments.begin(), _addedDocuments.end());
+  keys->erase(_removedDocuments.begin(), _removedDocuments.end());
```

Comment:
Err, this call to `erase` is probably wrong. The `erase` overload taking two iterators, unlike `insert`, expects to be given iterators belonging to the same instance of map. That is, the intended use case is something like:
```cpp
assert(foo < bar);
auto from = keys.find(foo);
auto to = keys.find(bar);
keys.erase(from, ++to);
```

Reply:
Done. I'm really not good at using iterators.